docs: update /contributing docs

meganesu opened the pull request:
## Description

This PR updates the docs under the /contributing part of the Gatsby site, to align with recent changes from the docs reorg project.

More specifically, this PR:

* Reorganizes files into subfolders, to make them easier to find.
* Updates the docs templates to match the new doc types: How-To Guides, Reference Guides, Conceptual Guides, and the Tutorial.

TODO:

- [ ] Add redirects for files that were moved or deleted.

LekoArts commented on docs/contributing/docs-contributions/how-to-write-a-plugin-readme.md:
```diff
@@ -3,6 +3,8 @@ title: How to Write a Plugin README
 issue: https://github.com/gatsbyjs/gatsby/issues/21599
```
Is this still needed?

meganesu replied:
Honestly, I left it because I'm not sure what this frontmatter does, and was mainly focused on content this time through 😅

meganesu replied:
Do you know what it's for? I don't mind deleting it!

LekoArts replied:
afaik it's for displaying a note about the stub issue. See at the bottom here: https://www.gatsbyjs.com/contributing/how-to-write-a-plugin-readme/

> See the issue relating to this stub on GitHub

So if we deleted all stubs then we can get rid of these `issue` in frontmatter

meganesu replied:
Gotcha, thanks! I'll delete it ✂️

meganesu replied:
(Hm that didn't work the way I thought it would. That's what I get for using the UI.)

LekoArts commented on docs/contributing/docs-contributions/how-to-write-a-reference-guide.md:
```diff
@@ -0,0 +1,77 @@
+
+Here are some general tips for helpful things to include:
+
+- Diagrams or other visuals, to show key processes or architectures.
```
Maybe call out that those diagrams should have a great alt text?

meganesu replied:
```suggestion
- Diagrams or other visuals, to show key processes or architectures.
  - Make sure to include alt text for accessibility! For
    help writing great alt text, refer to the W3C alt decision
    tree: https://www.w3.org/WAI/tutorials/images/decision-tree/
```

LekoArts commented on docs/contributing/stub-list.md:
```diff
@@ -1,10 +0,0 @@
----
```
If that was deleted, were all stubs references removed from the docs?

meganesu replied:
Good point. I just deleted and set up redirects for all the ones except the ones under "using gatsby professionally". (There's a different story to figure out what to do with those ones, since they should probably be pulled out of the docs site.)

LekoArts commented on docs/docs/how-to/plugins-and-themes/using-a-plugin-in-your-site.md:
```diff
@@ -1,56 +1,93 @@
 ---
-title: Using a Plugin in Your Site
+title: Add a Plugin to Your Site
```
Should this doc change be part of a separate PR?

meganesu replied:
I think it's okay to leave in here, since I updated it as part of the "how to write a how-to guide" contributing doc. To make something that fits the template.

LekoArts replied:
Okay 👍

LekoArts reviewed (changes requested):
Some smaller comments but overall great improvements!

meganesu commented on docs/contributing/docs-contributions/docs-structure.md:
```diff
@@ -0,0 +1,43 @@
+
+Sometimes, a single concept will require some combination of these documentation types. For example, the `gatsby-plugin-image` documentation includes the following pieces:
+
+- A Reference Guide to explain the plugin API and all its options.
+- A How-To Guide to show practical examples of how to use the plugin.
```
Great idea!

```suggestion
- [Gatsby Image plugin](/docs/reference/built-in-components/gatsby-plugin-image/): A Reference Guide to explain the plugin API and all its options.
- [Using the Gatsby Image plugin](/docs/how-to/images-and-media/using-gatsby-plugin-image/): A How-To Guide to show practical examples of how to use the plugin.
```

meganesu commented on docs/contributing/docs-contributions/how-to-write-a-conceptual-guide.md:
```diff
@@ -0,0 +1,30 @@
+
+## What is a Conceptual Guide?
+
+A Conceptual Guide provides a big-picture explanation of higher-level Gatsby concepts. Conceptual Guides are most useful for readers who want to build a deeper understanding of a particular topic.
```
I've been capitalizing them everywhere, like a proper noun. I was trying to show that it's the name of a specific thing, not just a general term. (#branding)

Does it feel off?

LekoArts replied:
It not necessarily felt "off" but seemed unusual as I'm only used to see this with "actual" brand names/proper nouns. But I don't have a strong opinion on this, feel free to keep it this way.

meganesu commented on docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md:
```diff
@@ -0,0 +1,88 @@
+
+- **Introduction:** Provide some quick background information to help readers understand 1) what this How-To Guide will help them achieve and 2) why that's useful.
+- **Prerequisites:** Any additional information or setup readers will have to do before they can make use of this How-To Guide.
+- **Directions:**
```
Oops! Thanks 😅

```suggestion
- **Directions:** List out the steps that readers to follow to complete the task. Use code blocks to show exactly what readers should type in their own projects.
```

meganesu commented on docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md:
```diff
@@ -0,0 +1,88 @@
+
+This section should be a step-by-step list of instructions for how readers
+can achieve the goal.
+
+1. Do the first thing.
+2. Do the next thing.
+
```
```suggestion

1. Do the first thing.
2. Do the next thing.

Use code blocks to show exactly what readers should type. The
Gatsby Style Guide includes tips for best practices when using code
blocks:
https://www.gatsbyjs.com/contributing/gatsby-style-guide/#format-code-blocks-inline-code-and-images

```

meganesu commented on docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md:
```diff
@@ -0,0 +1,88 @@
+Does the reader need to read another document first, install a particular
+plugin, or already know a certain skill? List those things here.
+
+## Directions
```
Good idea. Added a note below

meganesu commented on docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md:
```diff
@@ -0,0 +1,88 @@
+1. The first step of some other thing.
+1. The second step of some other thing.
+
+## Additional Resources
```
I don't have a super strong stance on this either way. My first thought is that we might be able to handle this on a case-by-case basis. Like if they have a really good article on their own blog, I don't mind if they include it.

Do you have a preference?

LekoArts replied:
Yeah, it'll be a case-by-case basis anyways. Maybe rephrase it then to mention that content _in the best case_ is evergreen (not quickly outdated), high quality, following our own CoC (strong language, a11y)

meganesu replied:
Can do! Added a line about that. (I left out the part about it being in English. We can add it in later if it ends up being a problem!)

meganesu commented on docs/contributing/blog-contributions.md:
```diff
@@ -2,7 +2,9 @@
-If you'd like to contribute a post to the Gatsby blog, please review the process and guidelines outlined below and submit your idea for the post to our [Gatsby blog proposal form](https://airtable.com/shr3449954866i3iF)
+Blog posts are a great way to share your own experiences working with Gatsby at work or on a personal project.
+
+If you'd like to contribute a post to the Gatsby blog, please review the process and guidelines outlined below and submit your post idea using the [Gatsby blog proposal form](https://airtable.com/shr3449954866i3iF).
```
I checked in with Michelle back in December and she said it still looked right. I'll ask her again just to be sure.

meganesu replied:
Sounds like she and Sam are revisiting the blog-specific process. I think it makes sense to add those updates in a future PR, if there are any.

LekoArts commented:
Now that the stubs are deleted, I think some issues here are/were the respective stub issues: https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22type%3A+documentation%22

meganesu commented on docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md:
```diff
@@ -0,0 +1,93 @@
+
+## Additional Resources
+
+Include other resources you think readers would benefit from or next steps
+they might want to take after reading your How-To Guide. You can also
+mention any resources that helped you write the article (blog posts, outside
+tutorials, etc.).
+
```
```suggestion
Include other resources you think readers would benefit from or next steps
they might want to take after reading your How-To Guide. You can also
mention any resources that helped you write the article (blog posts, outside
tutorials, etc.).

Best-case scenario, these should be high-quality, evergreen
(not quickly outdated) resources.
```

meganesu commented:
> Now that the stubs are deleted, I think some issues here are/were the respective stub issues: https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22type%3A+documentation%22

@LekoArts Good catch. I closed the issues.

I think I've addressed all the comments. Thanks for the review! Could you approve this? I'll handle coordinating merging this and the mansion PR.
